src/machines/child_workflow_state_machine.rs

```diff
@@ -241,30 +241,25 @@ impl Started {
         self,
         state: SharedState,
     ) -> ChildWorkflowMachineTransition<Cancelled> {
-        match state.cancellation_type {
-            ChildWorkflowCancellationType::Abandon => {
-                ChildWorkflowMachineTransition::ok(vec![], Cancelled::default())
-            }
```

Comment:
@bergundy Regarding this change, none of these cancel types can be handled within the child workflow machine.

Essentially they get handled at the lang SDK level (at least, how it works in Java now):
https://github.com/temporalio/sdk-java/blob/374014922d42b792a180ff5c8064e19819db0dc6/temporal-sdk/src/main/java/io/temporal/internal/statemachines/WorkflowStateMachines.java#L503

However, we could just upgrade the child workflow machine to deal with post-start cancels. It would eliminate lang having to deal with this logic, and it would simplify the cancellation of children from needing to decide if they should send the `CancelUnstartedChildWorkflowExecution` or the `RequestCancelExternalWorkflowExecution` command to just *always* sending a `CancelChildWorkflowExecution` command that always takes the seq number regardless of if the child is started or not.

LMK what you think about this. Maybe we do it another PR too. I think overall it would be a substantial simplification, but it'll take the better part of a day I think.